### DISCUSSION TITLE
[Support]: Multiple audio streams for WebRTC and MSE

### DISCUSSION THREAD

**Discussion by WeetbixNMilk** [blakeblackshear/frigate#15517] [General Support]
### Describe the problem you are having

Is it possible to have 2 audio tracks/codecs in a single stream and if so, will the client select the better stream from MSE or WebRTC?
The goal is to optimize compatibility.

In a perfect world, the solution would be, 
some_stream:
  -  rtsp://192.168.1.2/
  -  ffmpeg:some_stream#video=h264#audio=aac#audio=opus

Has anyone found a way to achieve this?

### Version

0.15.0-e76f4e9

### Frigate config file

```yaml
Not relevant.
```

### Relevant Frigate log output

```shell
Not relevant.
```

### Relevant go2rtc log output

```shell
Not relevant.
```

### FFprobe output from your camera

```shell
Not relevant.
```

### Frigate stats

```json
Not relevant.
```

### Install method

Docker Compose

### docker-compose file or Docker CLI command

```yaml
Not relevant.
```

### Object Detector

Coral

### Network connection

Wired

### Camera make and model

Not relevant.

### Screenshots of the Frigate UI's System metrics pages

Not relevant.

**Comment by NickM-27** [comment]
Yes, go2rtc will automatically choose the correct audio

**Reply by WeetbixNMilk** [reply]
Thanks Nick,

Currently using x2 "#audio=" commands crashes frigate.
I need to use x2 of these to support both MSE and WebRTC. My camera by default uses a weird codec which is supported by neither.

**Reply by NickM-27** [reply]
We'll need to see logs then. There's no reason that shouldn't work

**Reply by WeetbixNMilk** [reply]
When I asked chatGPT, this is the answer it provided.
I will recreate the logs now.

"Go2RTC does not natively support dual audio encoding within a single stream URL. The #audio parameter in the URL is designed to specify a single audio codec transformation for a given stream, and chaining multiple #audio parameters (#audio=aac#audio=opus) will not result in dual audio encoding but rather only apply the last #audio parameter in the chain.

Why It Doesn't Work

The #audio parameter modifies the audio encoding for the entire stream output. Since each transformation requires an independent ffmpeg process, Go2RTC doesn't provide a built-in way to include multiple audio streams (like AAC and Opus) in a single configuration.

To achieve dual audio encoding, you would need to handle this separately."

**Reply by NickM-27** [reply]
ChatGPT is wrong here

**Reply by hawkeye217** [reply]
From the [go2rtc docs](https://github.com/AlexxIT/go2rtc?tab=readme-ov-file#source-ffmpeg) for the ffmpeg source:

> You can use video and audio params multiple times (ex. `#video=copy#audio=copy#audio=pcmu`)
